javascript/ql/lib/ext/aws-sdk.model.yml

```
@@ -1,8 +1,52 @@
 extensions:
+  - addsTo:
+      pack: codeql/javascript-all
+      extensible: typeModel
+    data:
+      - ["AthenaClientV2", "aws-sdk", "Member[Athena]"]
+      - ["S3ClientV2", "aws-sdk", "Member[S3]"]
+      - ["RDSDataClientV2", "aws-sdk", "Member[RDSDataService]"]
+      - ["DynamoDBClientV2", "aws-sdk", "Member[DynamoDB]"]
```

```suggestion
      - ["aws-sdk.Athena", "aws-sdk", "Member[Athena]"]
      - ["aws-sdk.S3", "aws-sdk", "Member[S3]"]
      - ["aws-sdk.RDSDataService", "aws-sdk", "Member[RDSDataService]"]
      - ["aws-sdk.DynamoDB", "aws-sdk", "Member[DynamoDB]"]
```
Types should ideally be named after the correspinding TypeScript type if there is one, using the format `<package>.<typename>.`. This means we automatically match them up with type annotations and the names are also namespaced so different models easily clash with each other.

---

[d6329a9](https://github.com/github/codeql/pull/20135/commits/d6329a903c1bd5322e2afb2b12859f023e998984) + [3be5832](https://github.com/github/codeql/pull/20135/commits/3be5832b033cbf27e0d199723abae10f62cbdd80)